### PR TITLE
server: Avoid using "wayland-0" as WAYLAND_DISPLAY

### PR DESCRIPTION
**Pull request by Geo25rey**
Closes: #5687

See https://gitlab.freedesktop.org/wayland/weston/-/merge_requests/486

### CODE REVIEW

File: sway/server.c [hunk 1]
```diff
@@ -151,13 +152,30 @@ bool server_init(struct sway_server *server) {
...
+		}
+	}
+
 	if (!server->socket) {
 		sway_log(SWAY_ERROR, "Unable to open wayland socket");
 		wlr_backend_destroy(server->backend);
 		return false;
 	}
 
+	if (setenv("WAYLAND_DISPLAY", server->socket, true)) {
```
Comment (emersion) [inline sway/server.c:171]: The `setenv` already happens in `main.c` on line 390.

**Comment by panaman67** [comment]
If this is best practice, should a method be made in wlroots?

**Comment by Geo25rey** [comment]
> If this is best practice, should a method be made in wlroots?

That makes sense. Are you thinking adding a function wlr_renderer_init_wl_display_socket to render/wlr_renderer.c?

**Review by emersion — APPROVED** [review]
LGTM, thanks!

**Pull request merged**